lib/multicell_location/services/skyhook_integration.c

```diff
@@ -173,20 +175,22 @@ int location_service_generate_request(const struct lte_lc_cells_info *cell_data,
 		return err;
 	}
 
-	err = modem_info_string_get(MODEM_INFO_IMEI, imei, sizeof(imei));
-	if (err < 0) {
-		LOG_ERR("Failed to get IMEI, error: %d", err);
-		LOG_WRN("Falling back to uptime as user ID");
-
-		len = snprintk(imei, sizeof(imei), "%d", k_cycle_get_32());
-		if ((len < 0) || (len >= sizeof(imei))) {
-			LOG_ERR("Too small buffer for IMEI buffer");
-			return -ENOMEM;
+	if (device_id == NULL) {
+		err = modem_info_string_get(MODEM_INFO_IMEI, imei, sizeof(imei));
+		if (err < 0) {
+			LOG_ERR("Failed to get IMEI, error: %d", err);
+			LOG_WRN("Falling back to uptime as user ID");
+
+			len = snprintk(imei, sizeof(imei), "%d", k_cycle_get_32());
```

Comment:
Seems odd to me that Skyhook accepts a pseudorandom number for the id... I wonder how they use it.